core-primitives/stf-executor/src/enclave_signer.rs

```diff
@@ -0,0 +1,94 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{error::Result, traits::StfEnclaveSigning};
+use ita_stf::{AccountId, Index, KeyPair, Stf, TrustedCall, TrustedCallSigned};
+use itp_ocall_api::EnclaveAttestationOCallApi;
+use itp_sgx_crypto::{ed25519_derivation::DeriveEd25519, key_repository::AccessKey};
+use itp_stf_state_handler::handle_state::HandleState;
+use itp_types::ShardIdentifier;
+use sgx_externalities::SgxExternalitiesTrait;
+use sp_core::{ed25519::Pair as Ed25519Pair, Pair, H256};
+use std::sync::Arc;
+
+pub struct StfEnclaveSigner<OCallApi, StateHandler, ShieldingKeyRepository> {
+	state_handler: Arc<StateHandler>,
+	ocall_api: Arc<OCallApi>,
+	shielding_key_repo: Arc<ShieldingKeyRepository>,
+}
```

Comment:
The comment/documentation is there on the trait so far, not the implementation. I will add the second part of you suggestion. The first part IMO does not belong to this trait, since it's completely independent of the TOP-pool and not necessarily related.

Reply:
Hmm, ok I can understand that. However, this feature has mainly been discussed within the two of us. I want to prevent that this fragmented knowledge is hidden from the rest of the team. Do you have a suggestion on where to put this?

Reply:
The documentation is on the trait, where I added the additional info you suggested, so no knowledge should be lost?

Reply:
Ah wow sorry, I confused first part with second part. All good!